sim/battle.ts

```diff
@@ -2826,28 +2842,25 @@ export class Battle extends Dex.ModdedDex {
 			return false;
 		}
 
-		let p1switch = this.p1.active.some(mon => mon && !!mon.switchFlag);
-		let p2switch = this.p2.active.some(mon => mon && !!mon.switchFlag);
+		let switches = this.sides.map(side => side.active.some(mon => mon && !!mon.switchFlag));
```

Comment:
Two arrow functions in the same line is kind of unreadable

```suggestion
		let switches = this.sides.map(side =>
			side.active.some(mon => mon && !!mon.switchFlag)
		);
```